### PR TITLE
[PLUGIN-1719] Added ReadTimeout property in all salesforce plugins.

### PR DESCRIPTION
**Pull request by vikasrathee-cs**
[PLUGIN-1719] Added ReadTimeout property in all salesforce plugins.

[PLUGIN-1719]: https://cdap.atlassian.net/browse/PLUGIN-1719?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReader.java [hunk 1]
```diff
@@ -123,26 +123,28 @@ public boolean nextKeyValue() throws IOException {
...
-      } catch (AsyncApiException e) {
-        throw new IOException("Failed to query results", e);
       }
+      value = parserIterator.next().toMap();
+    } catch (AsyncApiException e) {
+      LOG.error("AsyncApiException error message with code is: {}" + e.getMessage());
+      throw new IOException("Failed to query results", e);
+    } catch (Exception e) {
+      throw new IOException("Failed to query results", e);
     }
```
Comment (MrRahulSharma) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReader.java:147]: Why this change to add a generic catch exception block ?

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceSourceConfigBuilder.java [hunk 2]
```diff
@@ -136,14 +137,20 @@ public SalesforceSourceConfigBuilder setConnectTimeout(Integer connectTimeout) {
...
     this.proxyUrl = proxyUrl;
     return this;
   }
 
   public SalesforceSourceConfig build() {
     return new SalesforceSourceConfig(referenceName, consumerKey, consumerSecret, username, password, loginUrl,
-                                      connectTimeout, query, sObjectName, datetimeAfter, datetimeBefore, duration,
+                                      connectTimeout, readTimeout, query, sObjectName, datetimeAfter, datetimeBefore,
+                                      duration,
                                       offset, schema, securityToken, operation, null, enablePKChunk,
```
Comment (MrRahulSharma) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceSourceConfigBuilder.java]:
```suggestion
                                      connectTimeout, readTimeout, query, sObjectName, datetimeAfter, datetimeBefore,
                                      duration, offset, schema, securityToken, operation, null, enablePKChunk,
```

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReader.java (hunk 1 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReader.java:147]: Added this to handle exceptions other than AsyncApiException.

File: docs/Salesforce-batchsink.md [hunk 3]
```diff
@@ -33,12 +33,15 @@ A Salesforce connected application must be created in order to get a client secr
 
 **Connect Timeout:** Maximum time in milliseconds to wait for connection initialization before it times out.
 
+**Read Timeout:** Maximum time in seconds to fetch data from the server before time out.
```
Comment (MrRahulSharma) [inline docs/Salesforce-batchsink.md]: Same comment in all the md files.

```suggestion
**Read Timeout:** Maximum time in seconds to wait for reading data from the server before it times out.
```

**Review by MrRahulSharma — APPROVED** [review]
